libr/debug/p/native/w32.c

```diff
@@ -442,81 +429,70 @@ static int debug_exception_event (DEBUG_EVENT *de) {
 }
 
 static char *get_file_name_from_handle (HANDLE handle_file) {
-	HANDLE handle_file_map;
-	CHAR *filename = NULL;
-
+	HANDLE handle_file_map = NULL;
+	LPTSTR filename = NULL;
 	DWORD file_size_high = 0;
+	LPVOID map = NULL;
 	DWORD file_size_low = GetFileSize (handle_file, &file_size_high);
+
 	if (file_size_low == 0 && file_size_high == 0) {
 		return NULL;
 	}
-
-	handle_file_map = CreateFileMappingA (handle_file, NULL, PAGE_READONLY, 0, 1, NULL);
-
+	handle_file_map = CreateFileMapping (handle_file, NULL, PAGE_READONLY, 0, 1, NULL);
 	if (!handle_file_map) {
-		return NULL;
+		goto err_get_file_name_from_handle;
 	}
-	filename = malloc(MAX_PATH+1);
-
-	/* Create a file mapping to get the file name. */
-	void* map = MapViewOfFile (handle_file_map, FILE_MAP_READ, 0, 0, 1);
-
-	if (!map) {
-		free (filename);
-		CloseHandle (handle_file_map);
-		return NULL;
+	filename = malloc ((MAX_PATH + 1) * sizeof (TCHAR));
+	if (!filename) {
+		goto err_get_file_name_from_handle;
 	}
-
-	if (!w32_GetMappedFileName (GetCurrentProcess (),
-		map,
-		filename,
-		MAX_PATH)) {
-
-		free(filename);
-		UnmapViewOfFile (map);
-		CloseHandle (handle_file_map);
-		return NULL;
+	/* Create a file mapping to get the file name. */
+	map = MapViewOfFile (handle_file_map, FILE_MAP_READ, 0, 0, 1);
+	if (!map || !w32_GetMappedFileName (GetCurrentProcess (), map, filename, MAX_PATH)) {
+		goto err_get_file_name_from_handle;
 	}
-
+	TCHAR temp_buffer[512];
 	/* Translate path with device name to drive letters. */
-	CHAR temp_buffer[512];
-	temp_buffer[0] = '\0';
-
-	if (!GetLogicalDriveStringsA (511, temp_buffer)) {
-		free (filename);
-		UnmapViewOfFile (map);
-		CloseHandle (handle_file_map);
-		return NULL;
+	if (!GetLogicalDriveStrings (sizeof (temp_buffer) - 1, temp_buffer)) {
+		goto err_get_file_name_from_handle;
 	}
-
-	CHAR name[MAX_PATH];
-	CHAR drive[3] =  " :";
-	BOOL found = FALSE;
-	CHAR *p = temp_buffer;
-	do {
+	TCHAR name[MAX_PATH];
+	TCHAR drive[3] =  TEXT (" :");
+	LPTSTR cur_drive = temp_buffer;
+	while (*cur_drive) {
 		/* Look up each device name */
-		*drive = *p;
-		if (QueryDosDeviceA (drive, name, MAX_PATH)) {
-			size_t name_length = strlen (name);
+		*drive = *cur_drive;
+		if (QueryDosDevice (drive, name, MAX_PATH)) {
```

Comment:
QueryDosDeviceW

Reply:
Comply with both windows calling convention and data types. UNICODE is already defined in file mk/platform.mk, TCHAR is wchar_t.